components/ParameterForm.vue

```diff
@@ -0,0 +1,236 @@
+              role="group"
+              :aria-label="parameter.label"
+              :size="largeScreen ? 'lg' : ''"
+            >
+              <CFormCheck
+                v-for="(option) in parameter.options"
+                :id="option.id"
+                :key="option.id"
+                v-model="formData[parameter.id]"
+                type="radio"
+                :button="{ color: 'primary', variant: 'outline' }"
+                :name="parameter.id"
+                autocomplete="off"
+                :label="option.label"
+                :value="option.id"
+              />
+            </CButtonGroup>
+          </CRow>
+        </CCol>
+        <div v-else>
+          <CIcon
+            v-if="icon(parameter)"
+            :icon="icon(parameter)"
+            class="parameter-icon"
+          />
+          <CFormSelect
+            :id="parameter.id"
+            v-model="formData[parameter.id]"
+            :label="parameter.label"
+            :aria-label="parameter.label"
+            :options="parameter.options.map((option: ParameterOption) => {
+              return { label: option.label, value: option.id };
+            })"
+            :size="largeScreen ? 'lg' : ''"
+          />
+        </div>
+      </div>
+      <div
+        v-if="globeParameter"
+        class="field-container"
+      >
+        <CIcon
+          v-if="icon(globeParameter)"
+          :icon="icon(globeParameter)"
+          class="parameter-icon"
+        />
+        <CFormSelect
+          :id="globeParameter.id"
+          v-model="formData[globeParameter.id]"
+          :data-test="`${globeParameter.id}-select`"
+          :label="globeParameter.label"
+          :aria-label="globeParameter.label"
+          :options="selectOptions(globeParameter)"
+          :size="largeScreen ? 'lg' : ''"
+        />
+      </div>
+      <CButton
+        id="run-button"
+        color="primary"
+        :size="largeScreen ? 'lg' : ''"
+        type="submit"
+      >
+        Run
+        <CIcon
+          icon="cilArrowRight"
+        />
+      </CButton>
+    </CForm>
+    <CAlert v-else-if="props.metadataFetchStatus === 'error'" color="warning">
+      Failed to retrieve metadata from R API. {{ metadataFetchError }}
+    </CAlert>
+    <CSpinner v-else-if="props.metadataFetchStatus === 'pending'" />
+  </div>
+</template>
+
+<script lang="ts" setup>
+import type { FetchError } from "ofetch";
+import { CIcon } from "@coreui/icons-vue";
+import type { MetaData, Parameter, ParameterOption } from "@/types/daedalusApiResponseTypes";
+import type { AsyncDataRequestStatus } from "#app";
+
+const props = defineProps<{
+  globeParameter: Parameter | undefined
+  metaData: MetaData | undefined
+  metadataFetchStatus: AsyncDataRequestStatus
+  metadataFetchError: FetchError | null
+}>();
+
+const formData = ref(
+  // Create a new object with keys set to the id values of the metaData.parameters array of objects, and all values set to refs with default values.
```

Comment:
This seems to work:

```ts
ref(
    props.metadata.parameters.reduce((accumulator, parameter) => {
      accumulator[parameter.id] = parameter.defaultOption || parameter.options[0].id;
      return accumulator;
    }, {} as { [key: string]: string | number }),
  )
```

And swapping out `ref`/`reactive` still passes all tests. So we can avoid having individual `ref`s as I had before while still wrapping the whole thing in a `ref()` instead of a `reactive()`. Personally, I prefer the explicitness of having to type `.value`, which reminds me when I'm using something that's reactive.